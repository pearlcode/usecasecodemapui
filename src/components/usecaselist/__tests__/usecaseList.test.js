/* eslint-disable react/jsx-filename-extension */
import { MemoryRouter as Router, Route, Link } from 'react-router-dom';
import {
    React,
    mount,
    shallowWithProps,
} from '../../app/__tests__/base';
import UsecaseList from '../index';
import correctData from '../../../__mockdata__/correctUsecases';

const usecaseListPath = '/usecases';

describe('UsecaseList ', () => {
    it('mounts properly with correct data', () => {
        const wrapper = shallowWithProps(UsecaseList, {
            usecases: correctData,
            match: { url: '/usecases' },
        });
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('.user-cases').length).toEqual(1);

        expect(wrapper.find('li').length).toEqual(correctData.length);

        expect(
            wrapper.find('.user-case__title').contains([correctData[0].title]),
        ).toEqual(true);

        expect(
            wrapper
                .find('.user-case__description')
                .contains([correctData[0].description]),
        ).toEqual(true);
    });

    it('has 2 links to usecase/:userId', () => {
        const wrapper = mount(
            <Router {...{ initialEntries: [usecaseListPath] }}>
                <Route
                    path={`${usecaseListPath}`}
                    render={({ match }) => (
                        <UsecaseList
                            {...{
                                match,
                                usecases: correctData,
                            }}
                        />
                    )}
                />
            </Router>,
            {
                usecases: correctData,
            },
        );
        const links = wrapper.find(Link);
        expect(links).toHaveLength(correctData.length);

        links.forEach((node, i) => {
            expect(node.props().to).toEqual(
                `${usecaseListPath}/${correctData[i].id}`,
            );
        });
    });
});
