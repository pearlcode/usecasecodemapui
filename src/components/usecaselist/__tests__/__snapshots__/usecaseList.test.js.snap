// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsecaseList  mounts properly with correct data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <usecaseList
    match={
      Object {
        "url": "/usecases",
      }
    }
    usecases={
      Array [
        Object {
          "description": "the first description",
          "id": 1,
          "title": "usecase 1",
        },
        Object {
          "description": "the second description",
          "id": 2,
          "title": "usecase 2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="user-case"
        >
          <div
            className="user-case__title"
          >
            <Link
              replace={false}
              to="/usecases/1"
            >
              usecase 1
            </Link>
          </div>
          <div
            className="user-case__description"
          >
            the first description
          </div>
        </li>,
        <li
          className="user-case"
        >
          <div
            className="user-case__title"
          >
            <Link
              replace={false}
              to="/usecases/2"
            >
              usecase 2
            </Link>
          </div>
          <div
            className="user-case__description"
          >
            the second description
          </div>
        </li>,
      ],
      "className": "user-cases",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="user-case__title"
            >
              <Link
                replace={false}
                to="/usecases/1"
              >
                usecase 1
              </Link>
            </div>,
            <div
              className="user-case__description"
            >
              the first description
            </div>,
          ],
          "className": "user-case",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/usecases/1"
              >
                usecase 1
              </Link>,
              "className": "user-case__title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "usecase 1",
                "replace": false,
                "to": "/usecases/1",
              },
              "ref": null,
              "rendered": "usecase 1",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "the first description",
              "className": "user-case__description",
            },
            "ref": null,
            "rendered": "the first description",
            "type": "div",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="user-case__title"
            >
              <Link
                replace={false}
                to="/usecases/2"
              >
                usecase 2
              </Link>
            </div>,
            <div
              className="user-case__description"
            >
              the second description
            </div>,
          ],
          "className": "user-case",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/usecases/2"
              >
                usecase 2
              </Link>,
              "className": "user-case__title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "usecase 2",
                "replace": false,
                "to": "/usecases/2",
              },
              "ref": null,
              "rendered": "usecase 2",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "the second description",
              "className": "user-case__description",
            },
            "ref": null,
            "rendered": "the second description",
            "type": "div",
          },
        ],
        "type": "li",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="user-case"
          >
            <div
              className="user-case__title"
            >
              <Link
                replace={false}
                to="/usecases/1"
              >
                usecase 1
              </Link>
            </div>
            <div
              className="user-case__description"
            >
              the first description
            </div>
          </li>,
          <li
            className="user-case"
          >
            <div
              className="user-case__title"
            >
              <Link
                replace={false}
                to="/usecases/2"
              >
                usecase 2
              </Link>
            </div>
            <div
              className="user-case__description"
            >
              the second description
            </div>
          </li>,
        ],
        "className": "user-cases",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="user-case__title"
              >
                <Link
                  replace={false}
                  to="/usecases/1"
                >
                  usecase 1
                </Link>
              </div>,
              <div
                className="user-case__description"
              >
                the first description
              </div>,
            ],
            "className": "user-case",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/usecases/1"
                >
                  usecase 1
                </Link>,
                "className": "user-case__title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "usecase 1",
                  "replace": false,
                  "to": "/usecases/1",
                },
                "ref": null,
                "rendered": "usecase 1",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "the first description",
                "className": "user-case__description",
              },
              "ref": null,
              "rendered": "the first description",
              "type": "div",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="user-case__title"
              >
                <Link
                  replace={false}
                  to="/usecases/2"
                >
                  usecase 2
                </Link>
              </div>,
              <div
                className="user-case__description"
              >
                the second description
              </div>,
            ],
            "className": "user-case",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/usecases/2"
                >
                  usecase 2
                </Link>,
                "className": "user-case__title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "usecase 2",
                  "replace": false,
                  "to": "/usecases/2",
                },
                "ref": null,
                "rendered": "usecase 2",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "the second description",
                "className": "user-case__description",
              },
              "ref": null,
              "rendered": "the second description",
              "type": "div",
            },
          ],
          "type": "li",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
